### DISCUSSION TITLE
Add `onError` hook

### DISCUSSION THREAD

**Discussion by feds01** [RobinTail/zod-sockets#244] [Ideas]
It would be nice to hook onto and handle situations where a particular action has a validation failure. Perhaps, this can be called `onValidationError`, and this allows the validation error to be intercepted and processed by the the application. For instance, a use case could look like:

```ts
export const config = createSimpleConfig({
    hooks: {
        onValidationError: async ({ client, error }) => {
              client.emit("error", transformErrorIntoMessage(error);
        },
        ...
    },

    emission: {
         error: {
            schema: z.tuple([ErrorMessageSchema]),
        },
     },
     ...
}
```

**Comment by RobinTail** [comment]
yeah, good idea

**Reply by RobinTail** [reply]
I did wanna make something like that, but then forgot about it. Thank you for noticing that gap

**Reply by RobinTail** [reply]
began working on it in #246

**Reply by RobinTail** [reply]
There will be a generic `onError` hook, but inside of it you can implement different logic depending of the error nature.
I will create multiple classes that extend from `Error`, similar to what I did in `express-zod-api`.

**Comment by RobinTail** [comment]
🚀 `onError` hook and `InputValidationError` are available in the recently released v2.1.0, @feds01 

Documentation: https://github.com/RobinTail/zod-sockets#handling-errors
